packages/gatsby/src/redux/reducers/jobsv2.ts

```diff
@@ -16,7 +23,9 @@ module.exports = (
 
     case `END_JOB_V2`: {
       const { jobContentDigest, result } = action.payload
-      const { job } = state.incomplete.get(jobContentDigest)
+      const { job } = state.incomplete.get(jobContentDigest) as IGatsbyJobV2
+
+      if (!job) return state
```

Comment:
We should actually throw an error here that says if we encountered this error it's a bug internal to Gatsby and to report an issue.

Reply:
@blainekasten @wardpeet Just to summarize the discussion. Should we get rid of this check, since the job would not be null never or should we trow an error? What you guys think is the best approach?

Reply:
throw an error :) Technically it could never happen but you a "Map" could return undefined so we should catch it

Reply:
I've just pushed some changes that I think that makes sense for both comments, fixing some types and statements inside the reducer. What you guys think about it?

Reply:
Great! I'll update it right now.

Reply:
Done! ✅